python/cudf/cudf/core/dataframe.py

```diff
@@ -356,7 +356,7 @@ def _init_from_series_list(self, data, columns, index):
             # Setting `final_columns` to self._index so
             # that the resulting `transpose` will be have
             # columns set to `final_columns`
-            self._index = final_columns
+            self._index = as_index(final_columns)
```

Comment:
To be honest I don't remember exactly what led to this. The problem is that `_get_union_of_indices` returns a column, so `final_columns` here is a column rather than an Index, which leads to the `transpose = self.T` line below failing. You need to convert the columns to an Index for the transpose to work. I don't recall exactly what change I made in this PR to trigger this failure, though it suggests that this line was somehow not being suitably tested before.

Reply:
If `final_columns` is known to be column here, would `._from_data` be more efficient to `as_index`?

Reply:
Unfortunately I don't think it's known to be a column; under some circumstances it can be an index, whereas under others it can be a column. That's why I used as_index.